From: Matt Kemp <matt@mattikus.com>
Date: Mon, 27 Feb 2017 14:03:23 -0600
Subject: Fix initramfs hook for merged /usr/lib and /lib

Under a merged `/lib` -> `/usr/lib` which renders `/lib` as a symlink,
`find /lib -type f -name libgcc_s.so.1` will not return a result as
`find` will not traverse the symlink. Modifying it to `find /lib/ -type
f -name libgcc_s.so.1` should work for both symlinked and non-symlinked
`/lib` directories.

Reviewed-by: Brian Behlendorf <behlendorf1@llnl.gov>
Signed-off-by: Matt Kemp <matt@mattikus.com>
Closes #5834
Applied-Upstream: https://github.com/zfsonlinux/zfs/commit/912e2ba92f45b61df476e7e15da33e0ea3eb0ae5
---
 contrib/initramfs/hooks/zfs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/contrib/initramfs/hooks/zfs b/contrib/initramfs/hooks/zfs
index 53e876d..dd796b6 100755
--- a/contrib/initramfs/hooks/zfs
+++ b/contrib/initramfs/hooks/zfs
@@ -55,7 +55,7 @@ mkdir -p "$DESTDIR/etc/"
 # automatically detected. The `find` utility and extended `cp` options are
 # used here because libgcc_s.so could be in a subdirectory of /lib for
 # multi-arch installations.
-cp --target-directory="$DESTDIR" --parents $(find /lib -type f -name libgcc_s.so.1)
+cp --target-directory="$DESTDIR" --parents $(find /lib/ -type f -name libgcc_s.so.1)
 
 for ii in $COPY_EXEC_LIST
 do
